src/validation.cpp

```diff
@@ -1345,9 +1345,14 @@ int GetSpendHeight(const CCoinsViewCache& inputs)
 
 
 static CuckooCache::cache<uint256, SignatureCacheHasher> scriptExecutionCache;
-static uint256 scriptExecutionCacheNonce(GetRandHash());
+static CSHA256 scriptExecutionCacheHasher;
```

Comment:
style-nit:
```suggestion
static CSHA256 g_scriptExecutionCacheHasher;
```

Also, should probably squash the third commit, so that all commits compile and pass the tests?

Reply:
I'd prefer to leave it intact as we would want to undo this commit in the future if we get rid of the 32 byte limit on GetRand which did not exist at the time this PR was made, but I'm happy to squash if it's your preference